seatunnel-connectors-v2/connector-qdrant/src/main/java/org/apache/seatunnel/connectors/seatunnel/qdrant/sink/QdrantBatchWriter.java

```diff
@@ -0,0 +1,188 @@
+                point.putPayload(fieldName, payloadValue);
+                continue;
+            }
+
+            Points.Vector vector = buildVector(fieldType, value);
+            if (vector != null) {
+                namedVectors.putVectors(fieldName, vector);
+            }
+        }
+
+        if (!point.hasId()) {
+            point.setId(id(UUID.randomUUID()));
+        }
+
+        point.setVectors(Points.Vectors.newBuilder().setVectors(namedVectors).build());
+        return point.build();
+    }
+
+    private void upsert() {
+        try {
+            qdrantClient
+                    .upsertAsync(
+                            Points.UpsertPoints.newBuilder()
+                                    .setCollectionName(collectionName)
+                                    .addAllPoints(qdrantDataCache)
+                                    .build())
+                    .get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException("Upsert failed", e);
+        }
+    }
+
+    public static Points.PointId pointId(SeaTunnelDataType<?> fieldType, Object value) {
+        SqlType sqlType = fieldType.getSqlType();
+        switch (sqlType) {
+            case INT:
+                return id(Integer.parseInt(value.toString()));
+            case STRING:
+                return id(UUID.fromString(value.toString()));
+            default:
+                throw new QdrantConnectorException(
+                        CommonErrorCode.UNSUPPORTED_DATA_TYPE,
+                        "Unexpected value type for point ID: " + sqlType.name());
+        }
+    }
+
+    public static JsonWithInt.Value buildPayload(SeaTunnelDataType<?> fieldType, Object value) {
+        SqlType sqlType = fieldType.getSqlType();
+        switch (sqlType) {
+            case SMALLINT:
+            case INT:
+            case BIGINT:
+                return ValueFactory.value(Integer.parseInt(value.toString()));
+            case FLOAT:
+            case DOUBLE:
+                return ValueFactory.value(Long.parseLong(value.toString()));
+            case STRING:
+            case DATE:
+                return ValueFactory.value(value.toString());
+            case BOOLEAN:
+                return ValueFactory.value(Boolean.parseBoolean(value.toString()));
+            default:
+                return null;
+        }
+    }
+
+    public static Points.Vector buildVector(SeaTunnelDataType<?> fieldType, Object value) {
+        SqlType sqlType = fieldType.getSqlType();
+        switch (sqlType) {
+            case FLOAT_VECTOR:
+            case FLOAT16_VECTOR:
+            case BFLOAT16_VECTOR:
+            case BINARY_VECTOR:
+                List<Float> floats = new ArrayList<>();
+                for (Object o : (Object[]) value) {
+                    floats.add(Float.parseFloat(o.toString()));
+                }
+                return VectorFactory.vector(floats);
+            default:
+                return null;
```

Comment:
Hi, 
1. the different vector type use different java class type to store data, not all use `List<Float>`.
```java
case FLOAT16_VECTOR:
case BFLOAT16_VECTOR:
case BINARY_VECTOR:
    (ByteBuffer)value
```
2. default should throw exception make sure user know the data type can not be stored.
3. Feel free to discuss our vector api data type, it is currently an experimental API, and we can modify it at any time to ensure that it works properly for all vector databases.

Reply:
Will do.

Reply:
Also pay attention to we update FLOAT_VECTOR from `List<Float>` to `ByteBuffer`. Please refer https://github.com/apache/seatunnel/pull/7446/files#diff-da88074adfe9dc704c385cb440c04c201ba810578ed7e1ea7c25c116277c8115R324

Reply:
Done. [`3f60035` (#7299)](https://github.com/apache/seatunnel/pull/7299/commits/3f6003566dd0e5fd65f6c130fec5d5fcb61f98a8)

The null returned by default is used to check if any vector exists. It isn't written into Qdrant.
https://github.com/Anush008/seatunnel/blob/3f6003566dd0e5fd65f6c130fec5d5fcb61f98a8/seatunnel-connectors-v2/connector-qdrant/src/main/java/org/apache/seatunnel/connectors/seatunnel/qdrant/sink/QdrantBatchWriter.java#L114-L117

Reply:
I am having problems running the integration tests for now.
```
Caused by: org.testcontainers.containers.ContainerFetchException: Failed to pull image: testcontainers/ryuk:0.3.4
```
I will have to sort that.